### PR TITLE
Bugfix/nijo 1167 fix error on cancelling bookings

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

File: Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs [hunk 1]
```diff
@@ -90,10 +94,43 @@ public Task<Result<BatchOperationResult>> Charge(List<int> bookingIds, ServiceAc
...
+                        await _bookingRecordManager.SetStatus(localBooking.ReferenceCode, BookingStatuses.ManualCorrectionNeeded);
+                        return Result.Failure<string>($"Booking {localBooking.ReferenceCode} with status {localBooking.Status} cannot be charged");
+                    }
+                }
+                
+                var chargeResult = await _accountPaymentService.Charge(localBooking, serviceAccount.ToUserInfo());
                 
                 if (chargeResult.IsFailure)
-                    await _bookingManagementService.Cancel(booking, serviceAccount.ToUserInfo());
+                    await _bookingManagementService.Cancel(localBooking, serviceAccount.ToUserInfo());
```
Comment (mingazhev) [inline Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs:133]: Why cancellation result is ignored?

File: Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs [hunk 2]
```diff
@@ -90,10 +94,43 @@ public Task<Result<BatchOperationResult>> Charge(List<int> bookingIds, ServiceAc
 
             async Task<Result<string>> Charge(Booking booking, UserInfo serviceAcc)
             {
-                var chargeResult = await _accountPaymentService.Charge(booking, serviceAccount.ToUserInfo());
+                var localBooking = booking;
+
+                if (localBooking.CheckInDate <= _dateTimeProvider.UtcNow())
+                {
+                    await _bookingRecordManager.SetStatus(localBooking.ReferenceCode, BookingStatuses.ManualCorrectionNeeded);
+                    return Result.Failure<string>($"Unable to charge for booking {localBooking.ReferenceCode}. Reason: check-in date expired");
```
Comment (mingazhev) [inline Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs:102]: check in, not check-in

File: Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs [hunk 3]
```diff
@@ -90,10 +94,43 @@ public Task<Result<BatchOperationResult>> Charge(List<int> bookingIds, ServiceAc
 
             async Task<Result<string>> Charge(Booking booking, UserInfo serviceAcc)
             {
-                var chargeResult = await _accountPaymentService.Charge(booking, serviceAccount.ToUserInfo());
+                var localBooking = booking;
```
Comment (mingazhev) [inline Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs:97]: Why this is needed? You may use booking from method parameters

**Review by mingazhev — CHANGES_REQUESTED** [review]

File: Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs (hunk 3 above)
Comment (DFSko) [reply Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs]: Fixed

File: Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs (hunk 1 above)
Comment (DFSko) [reply Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs]: Fixed

File: Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs (hunk 2 above)
Comment (DFSko) [reply Api/Services/Accommodations/Bookings/BatchProcessing/BookingsProcessingService.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**